File: spring-backend/src/main/java/com/suraj/smartgroceryapp/dto/AISuggestion.java
package com.suraj.smartgroceryapp.dto;


import com.suraj.smartgroceryapp.entity.GroceryCategory;
import com.suraj.smartgroceryapp.entity.Priority;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AISuggestion {
    private String name;
    private GroceryCategory category;
    private String reason;
    private Priority priority;
    private Integer quantity;
    private String unit;
}

File: smart-grocery-assistant/src/app/shared/services/grocery/ai-suggestion.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { GroceryItem, GroceryList } from '../../types/grocery.type';
import { environment } from '../../../../environments/environment.development';
interface Suggestion {
  name: string;
  category: string;
  reason: string;
  priority: 'low' | 'medium' | 'high';
  quantity?: number;
  unit?: string;
}
@Injectable({
  providedIn: 'root',
})
export class AiSuggestion {

  constructor(private http: HttpClient) {}

  getSuggestions(items: GroceryItem[]): Observable<{ suggestions: Suggestion[] }> {
    return this.http.post<{ suggestions: Suggestion[] }>(`${environment.expressUrl}/suggestions`, { items });
  }
}
